gym_sts/spaces/observations/components/combat.py

```diff
@@ -42,14 +42,22 @@ def __init__(self, state: dict):
             self.turn = combat_state["turn"]
 
             self.hand = [types.HandCard(**card) for card in combat_state["hand"]]
+
             self.discard = [types.Card(**card) for card in combat_state["discard_pile"]]
             self.discard.sort()
+
+            # TODO what if we have frozen eye? Then the draw order matters.
             self.draw = [types.Card(**card) for card in combat_state["draw_pile"]]
             self.draw.sort()
+
             self.exhaust = [types.Card(**card) for card in combat_state["exhaust_pile"]]
             self.exhaust.sort()
 
-            self.enemies = [types.Enemy(**enemy) for enemy in combat_state["monsters"]]
+            enemies = [types.Enemy(**enemy) for enemy in combat_state["monsters"]]
+            # CommunicationMod continues to send data for killed enemies/minions,
+            # so we need to filter them out.
+            self.enemies = [e for e in enemies if not e.is_gone]
```

Comment:
Did we make sure that the actions correspond to the correct enemies when this happens?

Reply:
Like, if enemy 1 is dead, enemies 2, 3, 4, .. would be mapped to 1, 2, 3, ... in the target of the PlayCard action

Reply:
Good catch. I looked into adjusting the target index of the `PlayCard` action right before submitting the action, and that's doable, but it's a bit hacky. It might be simpler to update our CommunicationMod fork to exclude enemies that are gone. Unsure if there's any value to considering them part of the game state.